mithril-common/src/store/limit_key_store.rs

```diff
@@ -0,0 +1,126 @@
+use async_trait::async_trait;
+use tokio::sync::RwLock;
+
+use super::{adapter::StoreAdapter, StoreError};
+
+/// Implementing this trait will make store able to limit the number of the
+/// stored records by pruning them &if a limit is set.
+#[async_trait]
+pub trait LimitKeyStore {
```

Comment:
Maybe you could have named it `StorePruner` which would be easier understand? WDYT?